### DISCUSSION TITLE
FieldDayLogger Discussions

### DISCUSSION THREAD

**Discussion by mbridak** [mbridak/FieldDayLogger#19] [Announcements]
## 👋 Welcome!

Hello, Most people I'd assume wont or don't use the issues system to post bugs or feature requests. It may seem too formal.
So, if you've used this in the past, I'd like to hear your suggestions, frustrations, general comments.

I know this will be a very low traffic area since the program is only used once a year. So I'm not expecting much.

73 Mike K6GTE

**Comment by msmcguire** [comment]
Hi, Mike. Thanks much for what looks like a great program. I'm hoping to use it for this weekend's Field Day (2023). I'm having trouble, though, with the Cloudlog integration. I've setup an API in Cloudlog for Field Day Logger. I've entered it in the Cloudlog API Key field and entered the Cloudlog URL, but I keep getting the following error:

WARNING __main__ - cloudlogauth Line 941:
Cloudlog: 404 Unable to authenticate.

Am I missing something simple? Any help or advice is much appreciated. Thanks.

73, Mike/K9ACM

**Comment by mbridak** [comment]
Hmm Odd. Unless he had made a breaking change to the API, I think you're
missing something. It should authenticate fine.

But there was a rather severe error in the posting code I found while
testing this for you. I've made the change and sent it to PyPi.
Should be v23.6.22

Thanks for the message.

Don't know if it would help at all, but the URL should end in api/

Mine is https://bond.zapto.org/Cloudlog/index.php/api/

**Reply by msmcguire** [reply]
Thanks, Michael, for your quick response. It seems to be authenticating now (after tweaking the URL), but it's not posting the QSOs to Cloudlog. The terminal is showing this error:

WARNING __main__ - cloudlogauth Line 927:
https://writing101.net/k9acm/cloudlog/index.php/api/auth/ [...]

(I removed the actual API key at the end of the URL above).

Is this a Cloudlog issue maybe?

Thanks again for your help.

Mike/K9ACM

**Comment by mbridak** [comment]
Hi, The warning message is not really a warning. It's just a debug message
set to the level of warning so it would be output even if debug messages
were turned off.

[image: Screenshot from 2023-06-23 10-25-45.png]
Here's one of those pictures you get when the developer goes, "well it's
working here..."

Which I know is totally not helpful... I don't have enough information to
know why it's not working there.

**Reply by msmcguire** [reply]
Thanks, Mike. I finally figured it out. I had to set the Cloudlog station profile ID in the fd_preferences.json file.

I'm having another problem, though. Now that FD is over, I'm trying to generate the logs. The cabrillo one looks fine, but the ADI file contains only 46 of the 154 contacts in the log. Not sure what's happening there. Any advice? I'm happy to provide any other details that might help troubleshoot this. 

Thanks again. Hope your FD was good.

73, Mike/K9ACM

**Reply by msmcguire** [reply]
Here is the output from the terminal if it helps at all...

Traceback (most recent call last):
  File "/home/mmcguire/.local/lib/python3.10/site-packages/fdlogger/__main__.py", line 2437, in generate_logs
    self.adif()
  File "/home/mmcguire/.local/lib/python3.10/site-packages/fdlogger/__main__.py", line 2212, in adif
    state = self.get_state(hissection)
  File "/home/mmcguire/.local/lib/python3.10/site-packages/fdlogger/__main__.py", line 2042, in get_state
    state = self.secState[section]
KeyError: 'NS'

**Reply by msmcguire** [reply]
I think it's choking on Nova Scotia (NS) for the section.

**Comment by msmcguire** [comment]
Okay, Mike, sorry for all the messages. I just wanted to let you know I figured it out. 'NS' was not in the arrl_sect.dat file. I added it, and it works. Also, I noticed that when if/when I enter a section incorrectly while logging, it allows me to proceed but then prevents the ADF file from generating fully (because the incorrect section is not in the arrl_sect.dat file). I wonder if the logger should prevent you from entering an incorrect section. 

Anyway, thanks for all your work on this. I look forward to using it again next year and will spread the word about your great program.

73,
Mike/K9ACM

**Comment by mbridak** [comment]
Hi, thanks for the heads up. I've pushed a fix to PyPi.
